src/market.py

```diff
@@ -0,0 +1,209 @@
+_MARKET_FORMAT += "Q"  # 8 bytes of baseLotSize
+_MARKET_FORMAT += "Q"  # 8 bytes of quoteLotSize
+_MARKET_FORMAT += "Q"  # 8 bytes of feeRateBps
+_MARKET_FORMAT += "7s"  # 7 bytes padding
+
+_MINT_LAYOUT = ""
+_MINT_LAYOUT += "36s"
+_MINT_LAYOUT += "b"
+_MINT_LAYOUT += "3s"
+
+
+ORDER_BOOK_LAYOUT = cStruct(Padding(5), "account_flag" / Padding(4), "slab_layout" / SLAB_LAYOUT)
+
+
+class MarketState(NamedTuple):
+    """Market State to stored the parsed market data."""
+
+    name: str
+    account_flags: int
+    ownAddress: str
+    vault_signer_nonce: int
+    base_mint: str
+    quote_mint: str
+    base_vault: str
+    base_deposits_total: int
+    base_fees_accrued: int
+    quote_vault: str
+    quote_deposits_total: int
+    quote_fees_accrused: int
+    quote_dust_threshold: int
+
+    request_queue: int
+    event_queue: int
+
+    bids: str
+    asks: str
+
+    base_lot_size: int
+    quote_lot_size: int
+
+    fee_rate_bps: int
+
+    padding: str
+
+    def get_initialized_flag(self):
+        """Return initialized account flag."""
+        # Not sure about this implementation, not sure why,
+        # it returns '\x03\x00\x00\x00\x00\x00\x00\x00', which means the 58th
+        # and 57th bits are set, however I was expecting 64th and 63th.
+        res = int.from_bytes(self.account_flags, "little") & 1
+        return res
+
+    def get_market_flag(self):
+        """Return isMarket account flag."""
+        res = (int.from_bytes(self.account_flags, "little") >> 1) & 1
+        return res
+
+
+class Market:
+    """Represents a Serum Market."""
+
+    _decode: Any
+    _baseSplTokenDecimals: int
+    _quoteSolTokenDecimals: int
+    _skipPreflight: bool
+    _confirmations: int
+    _porgram_id: PublicKey
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        decoded: MarketState,
+        base_mint_decimals: int,
+        quote_mint_decimals: int,
+        options: Any,  # pylint: disable=unused-argument
+        program_id: PublicKey = DEFAULT_DEX_PROGRAM_ID,
+    ):
+        # TODO: add options
+        if not decoded.get_initialized_flag or not decoded.get_market_flag:
+            raise Exception("Invalid market state")
+        self._decode = decoded
+        self._base_spl_token_decimals = base_mint_decimals
+        self._quote_spl_token_decimals = quote_mint_decimals
+        self._skip_preflight = False
+        self._confirmations = 10
+        self._program_id = program_id
+
+    @staticmethod
+    # pylint: disable=unused-argument
+    def load(endpoint: str, market_address: str, options: Any, program_id: PublicKey = DEFAULT_DEX_PROGRAM_ID):
+        """Factory method to create a Market."""
+        http_client = Client(endpoint)
+        base64_res = http_client.get_account_info(market_address)["result"]["value"]["data"][0]
+        res = Struct(_MARKET_FORMAT).unpack(base64.decodebytes(base64_res.encode("ascii")))
+        market_state = MarketState._make(res)
```

Comment:
For the alpha client, let's not try to deserialize in the most optimal manner. I think we should just write something that does a lot of slicing and get the `MarketState` tuple populated.

Here is the deserialize function I wrote Message:
https://github.com/michaelhly/solana-py/blob/0ad9e828608e1f988548a2a44e3f33b8cbf99de2/solana/message.py#L147

After getting everything working, let's come back and optimize for speed.